packages/babel-parser/src/plugins/flow.js

```diff
@@ -1461,6 +1524,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
           node.params = tmp.params;
           node.rest = tmp.rest;
+          node.this = tmp._this;
```

Comment:
Can you also update https://github.com/babel/babel/blob/d6816f0e6c233879788bcad8e47763f91e1b8776/packages/babel-parser/src/types.js#L161? Since we have a new `"this"` property in the AST node.

Reply:
The new `"this"` property is only present in the `FunctionTypeAnnotation` node. For regular function nodes the `"this"` is the first member of the parameter list (to match the Flow parser's behavior).

Reply:
This also matches the AST we use for `this` type annotations in TS.